src/Core/Configuration/OptionsConsumer.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.Extensions.Primitives;
+
+using System;
+
+namespace CodeMonkeys.Core.Configuration
+{
+    public abstract class OptionsConsumer<TOptions>
+        where TOptions : Options
+    {
+        protected IDisposable OptionsChangeToken;
+
+        protected OptionsConsumer(TOptions options)
+        {
+            OptionsChangeToken = ChangeToken.OnChange(
+                options.GetChangeToken,
+                OnOptionsHasChanged,
+                options);
+        }
+
+        protected abstract void OnOptionsHasChanged(TOptions options);
```

Comment:
prefer `OnOptionsChanged`

(`OnOptionsHasChanged` is grammatically wrong anyways)